src/Polly.Core/CircuitBreaker/CircuitBreakerStrategyOptions.TResult.cs

```diff
@@ -68,6 +69,14 @@ public class CircuitBreakerStrategyOptions<TResult> : ResilienceStrategyOptions
     [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "Addressed with DynamicDependency on ValidationHelper.Validate method")]
     public TimeSpan BreakDuration { get; set; } = CircuitBreakerConstants.DefaultBreakDuration;
 
+    /// <summary>
+    /// Gets or sets an optional delegate to use to dynamically generate the break duration.
+    /// </summary>
+    /// <value>
+    /// The default value is <see langword="null"/>.
+    /// </value>
+    public Func<BreakDurationGeneratorArguments, TimeSpan>? BreakDurationGenerator { get; set; }
```

Comment:
```suggestion
    public Func<BreakDurationGeneratorArguments, ValueTask<TimeSpan>>? BreakDurationGenerator { get; set; }
```

Polly v8 APIs should be natively async.

Reply:
The method OnActionFailureAsync is already asynchronous and utilizes the OpenCircuit_NeedsLock(Outcome<T> outcome, bool manual, ResilienceContext context, out Task? scheduledTask) method, which is not asynchronous. I'm having trouble grasping how to extract the assignment of _blockUntil from OpenCircuit_NeedsLock into another method. Can you provide a code example?

Reply:
Changing the `OpenCircuit_NeedsLock` to async is non-trivial as it involves locking which cannot be async. Since the generator is not invoked on a hot path I would just do sync over async with a comment to improve this in the future if someone complains. 

That way, we won't have to change the API if someone need async calculation of circuit breaker, rather we can just improve the implementation.

So let's just do this:

``` csharp

// Sync-over-async, improve this path in the future
if (_breakDurationGenerator is not null)
{
    breakDuration = _breakDurationGenerator(new(_behavior.FailureRate, _behavior.FailureCount, context)).GetAwaiter().GetResult();
}
```